app/(protected)/dashboard/clients/[clientId]/page.tsx

```diff
@@ -1,18 +1,229 @@
-import React from 'react';
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Card, CardContent } from "@/components/ui/card";
+import { toast } from "sonner";
+import { useRouter } from "next/navigation";
+
+
+import { getClientDetails, updateClient, deleteClient } from "@/app/(protected)/dashboard/clients/api";
+
 interface Props {
   params: Promise<{
     clientId: string;
   }>;
 }
-export default async function ClientDetailPage(props: Props) {
 
-  const {clientId} = await props.params
+interface Transaction {
+  id: string;
+  transactionDate: string;
+  description: string;
+  extractedReference?: string;
+  amount: number;
+}
+
+interface ClientWithTransactions {
+  id: string;
+  name: string;
+  clientReference: string;
+  expectedPaymentDay: number;
+  transactions: Transaction[];
+}
+
+export default function ClientDetailPage(props: Props) {
+  const [clientId, setClientId] = useState<string>("");
+  const [isEditing, setIsEditing] = useState(false);
+  const queryClient = useQueryClient();
+
+  useEffect(() => {
+    const resolveParams = async () => {
+      const { clientId } = await props.params;
+      setClientId(clientId);
+    };
+    resolveParams();
+  }, [props.params]);
+
```

Comment:
we really don't need a useEffect and useState here, we can extract the clientId directly and start using it. rule of thumb, avoid useEffect as much as you can😂. check this out for more info on how to extract the params https://nextjs.org/docs/app/guides/upgrading/version-15#params--searchparams

```suggestion
export default function ClientDetailPage(props: Props) {
  const {clientId} = use(props.params) // import use from "react" up
  const [isEditing, setIsEditing] = useState(false);
  const queryClient = useQueryClient();


```